### PR TITLE
add forkingClusterInvoker

### PR DESCRIPTION
**Pull request by xujianhai666**
**What this PR does**:

**Which issue(s) this PR fixes**:
add forkingClusterInvoker

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

File: common/constant/key.go [hunk 1]
```diff
@@ -48,6 +48,9 @@ const (
 	RETRIES_KEY          = "retries"
 	BEAN_NAME            = "bean.name"
 	FAIL_BACK_TASKS_KEY  = "failbacktasks"
+	FORKS_KEY            = "forks"
```
Comment (fangyincheng) [inline common/constant/key.go]: Pls add your new config to `config/testdata/` and test it by example.
image

Comment (xujianhai666) [reply common/constant/key.go]: @fangyincheng make an example? or add fork to testdata and then run success?

File: cluster/cluster_impl/forking_cluster_invoker.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+				selected = append(selected, ivk)
+			}
+		}
+	}
+
+	resultQ := queue.New(1)
+	for _, ivk := range selected {
+		go func(k protocol.Invoker) {
+			result := k.Invoke(invocation)
+			resultQ.Put(result)
```
Comment (hxmhlt) [inline cluster/cluster_impl/forking_cluster_invoker.go]: I think u should increase your judgment about error returned by resultQ.Put .

Comment (xujianhai666) [reply cluster/cluster_impl/forking_cluster_invoker.go]: done

File: common/constant/key.go (hunk 1 above)
Comment (xujianhai666) [reply common/constant/key.go]: @fangyincheng done

**Pull request merged**